FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpKey.kt

```diff
@@ -129,7 +129,11 @@ object PgpKey {
     oldPassphrase: Passphrase,
     newPassphrase: Passphrase
   ): PGPSecretKeyRing {
-    return encryptKey(decryptKey(key, oldPassphrase), newPassphrase)
+    return PGPainless.modifyKeyRing(key)
+      .changePassphraseFromOldPassphrase(oldPassphrase)
+      .withSecureDefaultSettings()
+      .toNewPassphrase(newPassphrase)
+      .done()
```

Comment:
As I understood from discussion, it requires change in the BC, and @vanitasvitae have posted issue to them, see https://github.com/pgpainless/pgpainless/issues/176#issuecomment-915136290

Reply:
It was a problem with `changeKeyPassphrase` method. Instead of using a single action(change passphrase), we used to use 2 actions(decrypt a key, encrypt a key with a new passphrase). These ways use different approaches. Paul has found an issue when using the second way. Switching to use the first way resolves our issue. But just for new users. If a user already has such a key(changed a passphrase with the app version > 1.1.8 and < 1.2.0) this fix will not help(I've checked). A user will receive an affected key again(because a new version of `changeKeyPassphrase` will produce a key that is based on the old key)

Reply:
Let me know if I should research into creating a method to change the S2K usage of existing keys to `SHA1`.

Reply:
@vanitaesvitae I suppose, when changing/settings pass phrase, we should always use the more secure mechanism regardless of what the key already has (if that makes sense)

Reply:
Makes sense. Moreover, now we absolutely need such method to let users fix keys which were affected.